app/main.py

```diff
@@ -1,13 +1,12 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"], "message": error["text"],
+            "name": error["code"], "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors], "path": file_path,
+            "status": "passed" if [format_linter_error(error) for error in errors] == [] else "failed"}
```

Comment:
The list comprehension `[format_linter_error(error) for error in errors]` is executed twice: once for creating the 'errors' list and once for checking if the list is empty to determine the 'status'. This can be inefficient for large lists. Consider storing the result in a variable and reusing it.